### DISCUSSION TITLE
Test mode = Alarm Trigger?

### DISCUSSION THREAD

**Discussion by elcid2012** [homebridge-simplisafe3/homebridge-simplisafe3#340] [Q&A] [answered]
I searched the bug reports and thought about filing one, but figured I would ask this question first.

Today I was debugging a sensor that had stopped responding and placed the BaseStation in test mode to do so. However, that results in an alarm getting triggered through HomeKit, which in turn runs some rather annoying automations and sends push notifications to everyone in the home (causing panic).

Is there a way that this plugin can delineate between test mode and actual alarm triggers? The normal HB log doesn't seem to suggest much, just gets an "Alarm received event: ALARM_TRIGGER"

Very understandable if this is not possible. SimpliSafe even calls you when you start testing devices, which is also annoying. At least that's now a pre-recorded message and not actually speaking with someone... anyway.

Thanks for the insights.

**Comment by shamoon** [comment, marked as answer]
Ah, this has annoyed me too. Actually there might be a way to sort this. Let me try some stuff

**Reply by elcid2012** [reply]
My man, thanks shamoon. Looking forward to what you come up with. Point me to the place I can buy you a coffee or beer.

**Reply by shamoon** [reply]
See https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/releases/tag/v1.10.5

No worries, I drink too much coffee anyway =)

**Reply by elcid2012** [reply]
Same, same.

So I installed 1.10.5 last night, was too busy to test anything.

Just now put my base station into test mode and HomeKit still sent a notification that the alarm was triggered (and of course SimpliSafe called me and sent me a push notification, wtf).

Anything I can provide for you to help debug?

**Reply by shamoon** [reply]
Odd. Just making sure you restarted?

You can add `this.log(data)` on line 517 of simplisafe.js, restart and then try again. In my testing 1.10.5 correctly detects this and just shows a warning in the logs.

You put it in test mode from the keypad?

**Reply by elcid2012** [reply]
Ah, yeah, a reboot did the trick. I see the warning in homebridge log as well. Thanks shamoon!
